### PR TITLE
Check for case where CSV Row has too many columns

### PR DESCRIPTION
**Pull request by prescod**
Better error checking for the case where a CSV Row has too many columns.

### CODE REVIEW

**Review by jofsky — APPROVED** [review]
Add docstrings!

File: snowfakery/standard_plugins/datasets.py [hunk 1]
```diff
@@ -113,12 +113,22 @@ def __init__(self, datasource: Path, repeat: bool):
...
+        self.results = (plugin_result(row) for row in d)
 
     def close(self):
         self.results = None
         self.file.close()
 
+    def plugin_result(self, row):
+        if None in row:
+            raise DataGenError(
+                f"Your CSV row has more columns than the CSV header:  {row[None]}, {self.datasource}"
```
Comment (davisagli) [inline snowfakery/standard_plugins/datasets.py:127]: Would be helpful to indicate which row had the problem

**Comment by prescod** [comment]
@*davisagli* : The row[None] stuff outputs the "extra values" so you can
search for them.

**Comment by davisagli** [comment]
@prescod Yeah I understand what it does, and it doesn't sound as helpful as a line number if the extra values happen to also appear in legitimate column positions.

**Comment by prescod** [comment]
I want to add as few instructions to the golden path as possible. It's a
pretty rare situation. I was already reluctant to add
additional instructions to the golden path to deal with it. You would have
really hated my first solution which was zero overhead but gave even less
context. I think this is a good middle ground between making every user pay
for the error handling and giving enough information for the error case to
be discoverable. If you load your CSV into Sheets or Excel, the extra row
should be visually obvious.

**Comment by davisagli** [comment]
It's hard to imagine that incrementing a counter would affect performance noticeably, but the concern with performance here seems fair.

**Pull request merged**